bolt/lib/Core/BinaryFunction.cpp

```diff
@@ -4472,6 +4472,18 @@ MCInst *BinaryFunction::getInstructionAtOffset(uint64_t Offset) {
   }
 }
 
+MCInst *BinaryFunction::getInstructionContainingOffset(uint64_t Offset) {
+  assert(CurrentState == State::Disassembled && "Wrong function state");
+
+  if (Offset > Size)
```

Comment:
interesting, is this something that can ever happen, or we want to make it an assertion too?

Reply:
> is this something that can ever happen

It depends on how we define the interface. If currently someone accidentally calls this with a large offset, they will get no instruction. Technically, this can happen when a function has a padding space that is considered a part of the function, but does not contain any instructions. That's why we have `Size` and `MaxSize` for `BinaryFunction`.

Reply:
Let's leave it like this for now, and maybe switch to an assertion later on.